### DISCUSSION TITLE
Class <Dialog> has no field doDialogue

### DISCUSSION THREAD

**Discussion by NikRetaNCAM** [HaxeFlixel/flixel#2460] [Q&A]
The dialog script (not finished)
![image](https://user-images.githubusercontent.com/60806448/145592174-c8155d36-7e42-463a-b8d0-1127f94dfa8f.png)

The code in MenuState.hx referring to that script
![image](https://user-images.githubusercontent.com/60806448/145592667-81b980bb-21c7-4c7d-b20f-c4fbc1f74d3b.png)

The error
![image](https://user-images.githubusercontent.com/60806448/145592722-6d0874c2-b9df-461e-b720-7e13d240af67.png)

How do i fix the thing???

**Comment by TheHENOOB** [comment]
Make the function static, if you want to call the function directly on a class
```
public static function doDialogue()
{

}
```

**Reply by NikRetaNCAM** [reply]
> Make the function static, if you want to call the function directly on a class
> 
> ```
> public static function doDialogue()
> {
> 
> }
> ```

then makeGraphic thing doesnt work (unknown identifier)

**Reply by TheHENOOB** [reply]
> > Make the function static, if you want to call the function directly on a class
> > 
> > ```
> > public static function doDialogue()
> > {
> > 
> > }
> > ```
> 
> then makeGraphic thing doesnt work (unknown identifier)

Make a variable with the dialog type on the state
```
var dialog:Dialog = new Dialog();
```
Remove the static keyword (my mistake), and call the function directly from the variable like this
```
dialog.doDialog();
```

**Comment by Geokureli** [comment]
Dialog extends FlxSprite, you don't control FlxSprites by saying FlxSprite.makeGraphic(), right? you create instances and call methods on them, like so:
```hx
var sprite = new FlxSprite();
sprite.makeGraphic(640, 480 0xbf);
```

The same goes for classes that extend FlxSprite, you create Instances and call functions on the instances:
```hx
var dialog = new Dialog();
dialog.doDialog();
```

**Reply by NikRetaNCAM** [reply]
I combined both of your things and it wor-
Edit: now, how do i add the sprite?
"add(sprite)" shows an unknown identifier
here's the cur. code in dialog.hx
![image](https://user-images.githubusercontent.com/60806448/145659262-24f7f84c-55d9-4bc6-b8ea-f504569ea67b.png)
and the other one
![image](https://user-images.githubusercontent.com/60806448/145659312-0d1ddb64-d943-4a1c-ad23-2f43178709be.png)

**Reply by Geokureli** [reply]
`add(dialog);`
same way you add a sprite. reminder that FlxSprite extenders are sprites and can be used just like a FlxSprite

**Reply by NikRetaNCAM** [reply]
> `add(dialog);` same way you add a sprite. reminder that FlxSprite extenders are sprites and can be used just like a FlxSprite

i tried that, and it told me that it's an unknown identifier
also i want to add blaccthing and dText, not dialog

**Reply by Geokureli** [reply]
`add(dialog)` goes in `clickPlay` because sprites can be added to groups or to states, and `clickPlay` is a function of a `FlxState`. sprites cannot be added to other `FlxSprite`, meaning you cannot add `blaccthing` to your Dialog instance. perhaps your Dialog class should extend `FlxGroup` so that you can add sprites to it.

I recommend following the flixel tutorial and learning the fundamentals a bit more before bringing every issue to the discussions

**Reply by NikRetaNCAM** [reply]
> `add(dialog)` goes in `clickPlay` because sprites can be added to groups or to states, and `clickPlay` is a function of a `FlxState`. sprites cannot be added to other `FlxSprite`, meaning you cannot add `blaccthing` to your Dialog instance. perhaps your Dialog class should extend `FlxGroup` so that you can add sprites to it.
> 
> I recommend following the flixel tutorial and learning the fundamentals a bit more before bringing every issue to the discussions

im doing it literally by official flixel tutorial (but with additions bc otherwise it'd be boring)

**Reply by Geokureli** [reply]
yeah but you seem to struggling with some pretty basic haxe stuff, if you follow the tutorial you should learn all the things you're trying to do. and check out some demos too to spice things up

**Comment by RichardBray** [comment]
@NicklGames Did you manage to fix this problem?
